examples/src-data-definition.md

```diff
@@ -0,0 +1,17 @@
+```python
```

Comment:
question: Material for MkDocs has a few nice features for [fenced code blocks](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#code-blocks). These require specifying configuration alongside the opening backticks.

I'm not sure if we can get this working with snex. There is some option of templating, but I don't think that's flexible enough to let us customise that some code blocks might have `\`\`\` py title="dsl.py"` and others `\`\`\`py hl_lines="4 5"`.

You can customise the language in the `# :snippet…` lines, but you can't use quotes as they get escaped.

If there's no workaround, is a better approach is to let the author add their own backticks? It's slightly more work for the author, but then allows for all the flexibility that Material for MkDocs offers.

Reply:
There's a similar question on still supporting [code annotations](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#code-annotations). It's easiest to see how this works in the [raw Markdown](https://raw.githubusercontent.com/squidfunk/mkdocs-material/b2ecd717bdebc71b514d3eb0ace10efc497fcf98/docs/reference/code-blocks.md) for those docs; see under "Code annotations".

In principle, this is simpler, because you could have a separate snippet, maybe inside a docstring inside the Python, that points to the annotation. But can we mark a snippet as "we don't want backticks". This also suggests having the author write backticks might be more flexible overall.

Reply:
You can change the output template for the files it produces, see the `output_template` config line here: https://github.com/jwbargsten/snex#advanced-snippet-tagging